third_party/move/move-vm/runtime/src/loader/script.rs

```diff
@@ -124,30 +118,24 @@ impl Script {
             .map(|tok| intern_type(BinaryIndexedView::Script(&script), tok, &struct_names))
             .collect::<PartialVMResult<Vec<_>>>()
             .map_err(|err| err.finish(Location::Undefined))?;
-        let return_ = Signature(vec![]);
-        let return_tys = return_
-            .0
-            .iter()
-            .map(|tok| intern_type(BinaryIndexedView::Script(&script), tok, &struct_names))
-            .collect::<PartialVMResult<Vec<_>>>()
-            .map_err(|err| err.finish(Location::Undefined))?;
-        let type_parameters = script.type_parameters.clone();
+        let ty_param_abilities = script.type_parameters.clone();
         // TODO: main does not have a name. Revisit.
         let name = Identifier::new("main").unwrap();
         let (native, def_is_native) = (None, false); // Script entries cannot be native
         let main: Arc<Function> = Arc::new(Function {
             file_format_version: script.version(),
             index: FunctionDefinitionIndex(0),
             code,
-            type_parameters,
+            ty_param_abilities,
             native,
             def_is_native,
             def_is_friend_or_private: false,
             scope,
             name,
-            return_types: return_tys.clone(),
-            local_types: local_tys,
-            parameter_types: parameter_tys.clone(),
+            // Script must not return values.
+            return_tys: vec![],
+            local_tys,
+            param_tys,
```

Comment:
Note: this should be fine as scripts cannot return values for now. We'll know to come back to this if something ends up getting broken.